doct.el

```diff
@@ -1282,6 +1283,87 @@ Otherwise, add to front."
   (let ((args (list (doct declarations) (copy-tree list))))
     (apply #'append (if append (nreverse args) args))))
 
+
+;; A structure for captures
+;; PARENTS is a list of strings which are the names of the capture's parents
+;; CHILD is t if the capture is a child of another and nil if it isn't
+;; DECLARATION is the declaration of the capture in doct
+(cl-defstruct doct--capture
+  parents
+  child
+  declaration)
```

Comment:
The reason behind the name `defcapture` was to have, as a user, a very clear mnemonic to access this very simple utility macro for defining capture templates. Since what you get by using the macro, from the perspective of the user, is a capture template, I wanted to call it `defcapture`. Just like `defun` is syntactic sugar for an operation like `(setq (symbol-function 'my-function) (lambda () ...))`. The machinery, including the declaration itself, wasn't meant to be seen by the user even in the name. Like in Common Lisp one doesn't see the calls to the constructors of the generic function funcallable object even though that's what happens when one uses `defgeneric`. It's all behind the scenes.

Since the purpose of the macro and it's functionality is only user ergonomics, even prepending `doct-` to the name is a bit of a bummer for me, to be completely honest. It makes me think that perhaps a better place to put it is in a configuration framework like Spacemacs or Doom emacs where user-facing utilities like Doom's `map!` do have a place.